rust/src/mime/detect.rs

```diff
@@ -60,3 +61,23 @@ pub unsafe extern "C" fn SCDetectMimeEmailGetUrl(
 
     return 0;
 }
+
+static mut HASH_BUF: Option<String> = None;
+#[no_mangle]
+pub unsafe extern "C" fn SCDetectMimeEmailGetBodyMd5(
+    ctx: &MimeStateSMTP, buffer: *mut *const u8, buffer_len: *mut u32,
+) -> u8 {
+    if ctx.md5_state == MimeSmtpMd5State::MimeSmtpMd5Completed {
+        HASH_BUF = Some(format!("{:x}", ctx.md5_result));
```

Comment:
This isn't safe, but I see how you landed here - need a place to put the formatted buffer. @catenacyber Got an idea here?  Thread local maybe?

Reply:
Make `HASH_BUF` part of `MimeStateSMTP`, maybe as an Option

See what other md5 like keywords do (quic cyu, hassh...)